extensions/search-regexp/src/components/ZipCodeList.tsx

```diff
@@ -0,0 +1,155 @@
+import { useEffect, useState } from "react";
+import got from "got";
+import { Icon, List, LocalStorage, showToast, Toast } from "@raycast/api";
+import { MappedExpression, ZipCodeResponse } from "../types";
+import { ExpressionItemActions } from "../searchRegexp";
+import { capitalizeSentence } from "../utilities";
+
+const ZIP_CODES_BASE_URL = "https://i18napis.appspot.com/address/data/";
+const ZIP_CODES_STORAGE_TOKEN = "ZIP_CODES";
+
+type ZipCodesMap = {
+  [key: string]: Record<string, string>;
+};
+
+enum Loading {
+  IDLE,
+  LOADING,
+  LOADED,
+}
+
+export default function ZipCodesList({ expressions }: { expressions: MappedExpression[] }): JSX.Element {
+  const [loading, setLoading] = useState<Loading>(Loading.IDLE);
+  const [zipCodes, setZipcodes] = useState<MappedExpression[] | null>(null);
+  const [filteredZipCodes, setFilteredZipCodes] = useState<MappedExpression[]>([]);
```

Comment:
when you have a lot of states in component, you can use [useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer)